### PR TITLE
Produce errors on ref-assignment to byval parameters

### LINKED ISSUE #28117: Ref-assigning to a non-ref parameter should be an error
```csharp
public class C {
    void M(int a, ref int b) {
        int local = 0;
        
        a = ref b;           // no error
        local = ref b;       // CS8373: The left-hand side of a ref assignment must be a ref local or parameter.
    }
}
```

### PR DESCRIPTION
**Pull request by OmarTawfik**
Fixes #28117

### CODE REVIEW

**Comment by OmarTawfik** [comment]
Note: add tests for val, in, and out.

**Comment by OmarTawfik** [comment]
@dotnet/roslyn-compiler for review.

File: src/Compilers/CSharp/Test/Semantic/Semantics/RefLocalsAndReturnsTests.cs [hunk 1]
```diff
@@ -3818,6 +3818,23 @@ void N()
...
+        [Fact]
+        [WorkItem(28117, "https://github.com/dotnet/roslyn/issues/28117")]
+        public void AssigningRefToParameter()
+        {
+            CreateCompilation(@"
+public class C
+{
+    void M(int a, ref int b)
+    {
+        a = ref b;
```
Comment (jaredpar) [inline src/Compilers/CSharp/Test/Semantic/Semantics/RefLocalsAndReturnsTests.cs]: Do we have an existing case which covers when the parameter is `in`?

Comment (OmarTawfik) [reply src/Compilers/CSharp/Test/Semantic/Semantics/RefLocalsAndReturnsTests.cs]: @jaredpar We have negative tests like `RefReturnRefExpression` in the same file above. The other cases lie under the escape safety feature and we have separate tests for these.
Is there another case you have in mind?


---
In reply to: [199555789](https://github.com/dotnet/roslyn/pull/28206#discussion_r199555789) [](ancestors = 199555789)

**Comment by OmarTawfik** [comment]
@dotnet-bot test windows_debug_spanish_unit32_prtest please
@dotnet-bot test windows_debug_vs-integration_prtest please
@dotnet-bot test windows_release_vs-integration_prtest please
// never triggered

**Comment by OmarTawfik** [comment]
@dotnet-bot test windows_debug_spanish_unit32_prtest please

**Pull request merged**